File: lib/go/factorial.go
package main

import "fmt"
func factorial(n int) int {
	if n == 0 {
		return 1
	}
	return n * factorial(n-1)
}
func main() {
        for i := 0; i < 1000000; i++ {
				fmt.Printf("%d\n", factorial(10))
        }
}

File: lib/test/cpp/factorial.cpp
#include <iostream>
int factorial(int x)
{
    if (x <= 0)
    {
        return 1;
    }
    return factorial(x - 1) * x;
}
int main()
{
    for (int i = 0; i < 1000000; i++)
    {
        std::cout << factorial(10) << std::endl;
    }

    return 0;
}